### PR TITLE
DAOS-3560 - Add slurm setup to ftest

### PR DESCRIPTION
**Pull request by mjean308**
Test-tag: soak_smoke

Signed-off-by: Maureen Jean <maureen.jean@intel.com>

### CODE REVIEW

File: ftest.sh [hunk 1]
```diff
@@ -304,6 +304,18 @@ if ${SETUP_ONLY:-false}; then
     exit 0
 fi
 
+# check if slurm needs to be configured for soak
+
+if [[ $TEST_TAG_ARG == *soak* ]]; then
+    IFS=, read -r -a CONTROL_NODE <<< $TEST_NODES
+    if ! ./slurm_setup.py -c ${CONTROL_NODE[0]} -n ${TEST_NODES} -s -i; then
```
Comment (brianjmurrell) [inline ftest.sh]: Does `slurrm_setup.py` need to be added in this PR?

File: ftest.sh [hunk 2]
```diff
@@ -304,6 +304,18 @@ if ${SETUP_ONLY:-false}; then
     exit 0
 fi
 
+# check if slurm needs to be configured for soak
+
+if [[ $TEST_TAG_ARG == *soak* ]]; then
+    IFS=, read -r -a CONTROL_NODE <<< $TEST_NODES
```
Comment (brianjmurrell) [inline ftest.sh]: Why not use the already parsed node list from lines 55-56?

File: ftest.sh (hunk 1 above)
Comment (mjean308) [reply ftest.sh]: yes.   when any soak tag is used; I would like to make sure that slurm is configured/enabled

File: ftest.sh (hunk 2 above)
Comment (mjean308) [reply ftest.sh]: I was going to use nodes but I need to use the first node in the TEST_NODES list.   I know I could use nodes[1] but if  the relation between nodes and TEST_NODES were to change then I may have to change the Control node as well.

File: ftest.sh (hunk 1 above)
Comment (brianjmurrell) [reply ftest.sh]: Ahh.  It has already been committed.

File: ftest.sh [hunk 3]
```diff
@@ -54,7 +54,7 @@ trap 'echo "encountered an unchecked return code, exiting with error"' ERR
 
 IFS=" " read -r -a nodes <<< "${2//,/ }"
 TEST_NODES=$(IFS=","; echo "${nodes[*]:1:8}")
-
+IFS="," read -r -a CONTROL_NODE <<< "${TEST_NODES}"
```
Comment (brianjmurrell) [inline ftest.sh]: I'm still not understanding why you are parsing `${TEST_NODES}` back into `$CONTROL_NODE` when you can just pick it out of `${nodes[]}`.  `${TEST_NODES}` is already derived from `${nodes}`.  I don't see what advantage there is to ignoring the parsing that has already been done and doing it again.

File: src/tests/ftest/container/simple_create_delete_test.py [hunk 4]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/python
-'''
-  (C) Copyright 2017-2019 Intel Corporation.
+"""
+  (C) Copyright 2018-2019 Intel Corporation.
```
Comment (brianjmurrell) [inline src/tests/ftest/container/simple_create_delete_test.py]: Why are we changing the copyright timeframe?

File: ftest.sh (hunk 3 above)
Comment (brianjmurrell) [reply ftest.sh]: I'm still not convinced of the need, but it's not a major issue.

File: src/tests/ftest/container/simple_create_delete_test.py (hunk 4 above)
Comment (mjean308) [reply src/tests/ftest/container/simple_create_delete_test.py]: visual studio linting error.  pydocstyle(D300)  I usually change it to clear out any linting issues before pushing change

File: ftest.sh [hunk 5]
```diff
@@ -304,6 +304,16 @@ if ${SETUP_ONLY:-false}; then
     exit 0
 fi
 
+# check if slurm needs to be configured for soak
+if [[ \"${TEST_TAG_ARG}\" =~ soak ]]; then
+    if ! ./slurm_setup.py -c ${nodes[0]} -n ${TEST_NODES} -s -i; then
+        rc=\${PIPESTATUS[0]}
+        exit \"$rc\"
```
Comment (brianjmurrell) [inline ftest.sh]: Don't really need to use `$rc` here.  You could just do:

```
exit \${PIPESTATUS[0]}
```

File: src/tests/ftest/util/io_utilities.py [hunk 6]
```diff
@@ -327,86 +326,6 @@ def read_array_objects(container, size, items, dkey, akey, obj, txn):
     return [item[:-1] for item in data]
 
 
-def read_during_rebuild(
```
Comment (saurabhtandan) [inline src/tests/ftest/util/io_utilities.py]: Any specific reason these are removed?

File: src/tests/ftest/util/general_utils.py [hunk 7]
```diff
@@ -244,190 +242,6 @@ def get_random_string(length, exclude=None):
     return random_string
 
 
-@fail_on(DaosApiError)
-def get_pool(context, mode, size, name, svcn=1, log=None, connect=True):
```
Comment (saurabhtandan) [inline src/tests/ftest/util/general_utils.py]: Any specific reason these are removed?

File: src/tests/ftest/soak/soak.py [hunk 8]
```diff
@@ -134,7 +134,7 @@ def create_ior_cmdline(self, job_params, job_spec, pool):
         """
         command = []
         iteration = self.test_iteration
-        ior_params = "/run/" + job_spec + "/"
+        ior_params = "/run/" + job_spec + "/*"
```
[the remark this answers is not in the document]
Comment (mjean308) [reply src/tests/ftest/soak/soak.py]: changed this but will updated soak in PR-2511 to get rid of the other instances or "+"

File: src/tests/ftest/util/io_utilities.py (hunk 6 above)
Comment (mjean308) [reply src/tests/ftest/util/io_utilities.py]: The methods io_utils that have been removed have been updated with code that resides in test_utils now and this routine was not used anywhere and was replaced with test_utils,read_data_during_rebuild

File: src/tests/ftest/util/general_utils.py (hunk 7 above)
Comment (mjean308) [reply src/tests/ftest/util/general_utils.py]: This also was not use any were and has been replaced with test_utils.pool

**Comment by brianjmurrell** [comment]
How come [master runs 1511 tests](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/master/1277/testReport/) and this PR [only runs 311](https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-1328/35/testReport/(root)/) when master runs the same set of tags PRs are supposed to run?

**Comment by johannlombardi** [comment]
Please address the conflict.

**Pull request merged**